docs/proposals/config_data.md

```diff
@@ -0,0 +1,316 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named `ConfigData` stored inside `etcd` that
```

Comment:
A very apt comment. There is no reason for `ConfigData` to be linked to `etcd`. Also `ConfigData` being used with a separate k/v storage can be very helpful where configuration blobs are huge, but this is a possible improvement out of scope for this proposal.